hadoop-ozone/fault-injection-test/mini-chaos-tests/src/test/java/org/apache/hadoop/ozone/MiniOzoneLoadGenerator.java

```diff
@@ -44,66 +39,99 @@
   private static final Logger LOG =
       LoggerFactory.getLogger(MiniOzoneLoadGenerator.class);
 
-  private final List<LoadExecutors> loadExecutors;
+  private final List<LoadGenerator> loadGenerators;
+  private final LoadExecutors loadExecutor;
 
   private final OzoneVolume volume;
   private final OzoneConfiguration conf;
   private final String omServiceID;
 
-  MiniOzoneLoadGenerator(OzoneVolume volume, int numClients, int numThreads,
-      int numBuffers, OzoneConfiguration conf, String omServiceId)
+  MiniOzoneLoadGenerator(OzoneVolume volume, int numThreads,
+      int numBuffers, OzoneConfiguration conf, String omServiceId,
+      List<Class<? extends LoadGenerator>> loadGenratorClazzes)
       throws Exception {
     DataBuffer buffer = new DataBuffer(numBuffers);
-    loadExecutors = new ArrayList<>();
+    loadGenerators = new ArrayList<>();
     this.volume = volume;
     this.conf = conf;
     this.omServiceID = omServiceId;
 
-    // Random Load
-    String mixBucketName = RandomStringUtils.randomAlphabetic(10).toLowerCase();
-    volume.createBucket(mixBucketName);
-    List<LoadBucket> ozoneBuckets = new ArrayList<>(numClients);
-    for (int i = 0; i < numClients; i++) {
-      ozoneBuckets.add(new LoadBucket(volume.getBucket(mixBucketName),
-          conf, omServiceId));
+    for(Class<? extends LoadGenerator> clazz : loadGenratorClazzes) {
+      addLoads(clazz, buffer);
     }
-    RandomLoadGenerator loadGenerator =
-        new RandomLoadGenerator(buffer, ozoneBuckets);
-    loadExecutors.add(new LoadExecutors(numThreads, loadGenerator));
 
-    // Aged Load
-    addLoads(numThreads,
-        bucket -> new AgedLoadGenerator(buffer, bucket));
-
-    //Filesystem Load
-    addLoads(numThreads,
-        bucket -> new FilesystemLoadGenerator(buffer, bucket));
-
-    //Repl Load
-    addLoads(numThreads,
-        bucket -> new ReadOnlyLoadGenerator(buffer, bucket, 20));
+    this.loadExecutor = new LoadExecutors(numThreads, loadGenerators);
   }
 
-  private void addLoads(int numThreads,
-                        Function<LoadBucket, LoadGenerator> function)
-      throws Exception {
+  private void addLoads(Class<? extends LoadGenerator> clazz,
+                        DataBuffer buffer) throws Exception {
     String bucketName = RandomStringUtils.randomAlphabetic(10).toLowerCase();
     volume.createBucket(bucketName);
-    LoadBucket bucket = new LoadBucket(volume.getBucket(bucketName), conf,
-        omServiceID);
-    LoadGenerator loadGenerator = function.apply(bucket);
-    loadExecutors.add(new LoadExecutors(numThreads, loadGenerator));
+    LoadBucket ozoneBucket = new LoadBucket(volume.getBucket(bucketName),
+        conf, omServiceID);
+
+    LoadGenerator loadGenerator = clazz
+        .getConstructor(DataBuffer.class, LoadBucket.class)
+        .newInstance(buffer, ozoneBucket);
```

Comment:
Can you please document the required signature for the constructors of `LoadGenerator` implementations in the interface?

Reply:
done, great comment :)